src/core/service.c

```diff
@@ -1121,14 +1124,17 @@ static usec_t service_coldplug_timeout(Service *s) {
                 return usec_add(UNIT(s)->active_enter_timestamp.monotonic, s->runtime_max_usec);
 
         case SERVICE_STOP:
-        case SERVICE_STOP_WATCHDOG:
         case SERVICE_STOP_SIGTERM:
         case SERVICE_STOP_SIGKILL:
         case SERVICE_STOP_POST:
         case SERVICE_FINAL_SIGTERM:
         case SERVICE_FINAL_SIGKILL:
                 return usec_add(UNIT(s)->state_change_timestamp.monotonic, s->timeout_stop_usec);
 
+        case SERVICE_STOP_WATCHDOG:
+                return usec_add(UNIT(s)->state_change_timestamp.monotonic,
+                                s->timeout_abort_usec > 0 ? s->timeout_abort_usec : s->timeout_stop_usec);
```

Comment:
i think having a tiny helper call like this would make sense:

```c
static usec_t service_timeout_abort_usec(Service *s) {
        return s->timeout_abort_usec > 0 ? s->timeout_abort_usec : s->timeout_stop_usec;
}
```

after all, you are currently using the same ternary expression at multiple times now, hence I think it would be nicer to have just one implementation of it.